contractcourt/htlc_timeout_resolver.go

```diff
@@ -374,12 +381,35 @@ func (h *htlcTimeoutResolver) Resolve() (ContractResolver, error) {
 		if err := waitForOutputResolution(); err != nil {
 			return nil, err
 		}
+
+		// Once our success tx has confirmed, we add a resolution for
+		// our success tx first stage transaction.
+		successTx := h.htlcResolution.SignedTimeoutTx
+		spendHash := successTx.TxHash()
+
+		reports = append(reports, &channeldb.ResolverReport{
+			OutPoint:        successTx.TxIn[0].PreviousOutPoint,
+			Amount:          h.htlc.Amt.ToSatoshis(),
+			ResolverType:    channeldb.ResolverTypeOutgoingHtlc,
+			ResolverOutcome: channeldb.ResolverOutcomeFirstStage,
+			SpendTxID:       &spendHash,
+		})
 	}
 
 	// With the clean up message sent, we'll now mark the contract
-	// resolved, and wait.
+	// resolved, record the timeout and the sweep txid on disk, and wait.
 	h.resolved = true
-	return nil, h.Checkpoint(h)
+
+	amt := btcutil.Amount(h.htlcResolution.SweepSignDesc.Output.Value)
+	reports = append(reports, &channeldb.ResolverReport{
+		OutPoint:        h.htlcResolution.ClaimOutpoint,
+		Amount:          amt,
+		ResolverType:    channeldb.ResolverTypeOutgoingHtlc,
+		ResolverOutcome: channeldb.ResolverOutcomeTimeout,
```

Comment:
Still struggling with the definition of those outcomes. Inside the context of an outgoing htlc, we can only get it or not get it.